test/version_solver_test.dart

```diff
@@ -638,6 +659,27 @@ void badSource() {
     await d.appDir({'foo': 'any'}).create();
     await expectResolves(result: {'foo': '1.0.0', 'bar': '1.0.0'}, tries: 2);
   });
+
+  // Issue 1853
+  test('reports a nice error across a collapsed cause', () async {
+    await servePackages((builder) {
+      builder.serve('foo', '1.0.0', deps: {'bar': 'any'});
+      builder.serve('bar', '1.0.0', deps: {'baz': 'any'});
+      builder.serve('baz', '1.0.0');
+    });
+    await d.dir('baz', [d.libPubspec('baz', '1.0.0')]).create();
+
+    await d.appDir({
+      'foo': 'any',
+      'baz': {'path': '../baz'}
+    }).create();
+    await expectResolves(error: equalsIgnoringWhitespace('''
+      Because every version of foo depends on bar any which depends on baz any,
```

Comment:
The "any" doesn't seem to add much value here and makes it a little harder to read. What do you think of just showing the package name if the constraint includes all versions?

> Because every version of foo depends on bar which depends on baz, every version of foo requires baz from hosted.
> So, because myapp depends on both baz from path and foo, version solving failed.

?

Reply:
I considered that, but I think it makes sense to reproduce the constraints from the packages' pubspecs here even if they're `any`. Just saying `bar depends on baz` risks making it unclear that there's no other version of `bar` that could work.

Also, since we discourage `any` constraints in real packages, this is unlikely to come up very often in practice.